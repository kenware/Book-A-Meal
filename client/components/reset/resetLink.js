import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';

import Header from '../header/index';
import Footer from '../footer/index';
import * as actions from '../../redux/Action/action';


class ResetLink extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      emailOrUsername: '',
      submit: 'Submit'
    };
    this.onChange = this.onChange.bind(this);
    this.resetLink = this.resetLink.bind(this);
  }
  /**
   * lifecycle hook called when component receives props
   *
   * @memberof Dashboard index
   * return state after it is changed
   */
  componentWillReceiveProps(newProps) {
    if (newProps) {
      this.setState({ submit: 'Submit' });
    }
  }
  /**
   * @param  {} e set state on input onchange event
  */
  onChange(e) {
    const { state } = this;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }
  /**
   * calls redux action
   * sends resetlink to user email
  */
  resetLink(e) {
    e.preventDefault();
    const {
      emailOrUsername,
    } = this.state;

    if (emailOrUsername) {
      this.props.actions.resetLink(emailOrUsername);
      this.setState({
        submit: (<div><i className="fa fa-spinner fa-spin fa-2x fa-fw" aria-hidden="true" /></div>)
      });
      this.setState({ inputError: '' });
      return;
    }
    this.setState({ inputError: 'this field is required' });
  }

  render() {
    return (
      <div>
        <Header />
        <div className="register-container">
          <div className="register-col" />
          <div className="register-wrapper">
            <h2 className="login-header">Enter your username or email</h2>
            <h3 className="login-header danger text-center">{this.props.errorMessage.resetError}</h3>
            <h3 className="login-header p-color text-center">{this.props.successMessage.resetSuccess}</h3>
            <form action="dashboard.html">
              <div className="form-field">
                <label htmlFor="name">Username/Email <br />
                  <span className="danger">{this.state.inputError} </span>
                </label>
                <input onChange={this.onChange} type="text" id="username" name="emailOrUsername" placeholder="keny or keny@gmail.com" />
              </div>
              <div className="form-field">
                <button type="submit" className="button lg" onClick={this.resetLink}>{this.state.submit}</button>
              </div>
              <div className="form-field">
                <label htmlFor="inputPassword3">
                  <Link to="/login">Login</Link>&nbsp;&nbsp;&nbsp;&nbsp;
                  <Link to="/register">Register?</Link>
                </label>
              </div>
            </form>

          </div>
          <div className="register-col" />
        </div>
        <Footer />
      </div>

    );
  }
}
ResetLink.propTypes = {
  errorMessage: PropTypes.object.isRequired,
  successMessage: PropTypes.object.isRequired,
  actions: {
    clearMessages: PropTypes.func.isRequired,
    resetLink: PropTypes.func.isRequired
  }.isRequired,
};

function mapStateToProps(state) {
  return {
    errorMessage: state.errorMessage,
    successMessage: state.successMessage,
  };
}
function mapDispatchToProps(dispatch) {
  return { actions: bindActionCreators(actions, dispatch) };
}
export default connect(mapStateToProps, mapDispatchToProps)(ResetLink);
