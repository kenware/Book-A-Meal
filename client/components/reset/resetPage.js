import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import Header from '../header/index';
import Footer from '../footer/index';
import * as actions from '../../redux/Action/action';

export class ResetPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      password: '',
      submit: 'Submit'
    };
    this.onChange = this.onChange.bind(this);
    this.newPassword = this.newPassword.bind(this);
  }
  /**
   * lifecycle hook called when component receives props
   *
   * @memberof Dashboard index
   * return state after it is changed
   */
  componentWillReceiveProps(newProps) {
    if (newProps) {
      this.setState({ submit: 'Submit' });
    }
  }
  /**
   * @param  {} e set state on input onchange event
  */
  onChange(e) {
    const { state } = this;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }
  /**
   * calls redux action
   * sends resetlink to user email
  */
  newPassword(e) {
    e.preventDefault();
    const { password } = this.state;
    const { token } = this.props.match.params;
    if (password) {
      this.props.actions.changePassword(password, token);
      this.setState({
        submit: (<div><i className="fa fa-spinner fa-spin fa-2x fa-fw" aria-hidden="true" /></div>)
      });
      this.setState({ inputError: '' });
      return;
    }
    this.setState({ inputError: 'this field is required' });
  }

  render() {
    return (
      <div>
        <Header />
        <div className="register-container">
          <div className="register-col" />
          <div className="register-wrapper">
            <h2 className="login-header">Enter Your New Password</h2>
            <h3 className="login-header danger text-center">{this.props.errorMessage.passwordResetError}</h3>
            <h3 className="login-header p-color text-center">{this.props.successMessage.passwordSuccess}</h3>
            <form action="dashboard.html">
              <div className="form-field">
                <label htmlFor="name">New Password <br />
                  <span className="danger">{this.state.inputError} </span>
                </label>
                <input onChange={this.onChange} type="password" id="password" name="password" placeholder="124gdt446776j7u" />
              </div>
              <div className="form-field">
                <button type="submit" className="button lg" onClick={this.newPassword}>{this.state.submit}</button>
              </div>
              <div className="form-field">
                <label htmlFor="inputPassword3">
                  <Link to="/login">Login</Link>&nbsp;&nbsp;&nbsp;&nbsp;
                  <Link to="/register">Register?</Link>
                </label>
              </div>
            </form>

          </div>
          <div className="register-col" />
        </div>
        <Footer />
      </div>

    );
  }
}
ResetPage.propTypes = {
  match: PropTypes.object.isRequired,
  errorMessage: PropTypes.object.isRequired,
  successMessage: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired,
};

export const mapStateToProps = state => ({
  errorMessage: state.errorMessage,
  successMessage: state.successMessage,
});

export const mapDispatchToProps = dispatch => ({ actions: bindActionCreators(actions, dispatch) });
export default connect(mapStateToProps, mapDispatchToProps)(ResetPage);
