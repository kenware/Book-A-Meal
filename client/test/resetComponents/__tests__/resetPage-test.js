import React from 'react';
import { shallow } from 'enzyme';
import { ResetPage, mapStateToProps, mapDispatchToProps } from '../../../components/reset/resetPage';
import { props, emptyProps } from '../resetMockData';

describe('Test ResetLink component', () => {
  it('renders correctly', () => {
    const tree = shallow(<ResetPage {...emptyProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render messages from props', () => {
    const tree = shallow(<ResetPage {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to change event and change the state of the resetLink Component', () => {
    const tree = shallow(<ResetPage {...props} />);
    const wrapper = tree.instance();
    wrapper.newPassword({ preventDefault: jest.fn() });
    tree.find('#password').simulate('change', { target: { name: 'password', value: 'kevin234355' } });
    wrapper.newPassword({ preventDefault: jest.fn() });
    expect(tree).toMatchSnapshot();
  });

  it('should respond to componentWillRecievProps methods', () => {
    const tree = shallow(<ResetPage {...props} />);
    const wrapper = tree.instance();
    wrapper.componentWillReceiveProps();
    wrapper.componentWillReceiveProps({ successMessage: 'Password changed' });
    expect(tree).toMatchSnapshot();
  });

  it('should respond to mapStateToProps methods', () => {
    const ownProps = { match: { params: { mealId: 1 } } };
    const tree = mapStateToProps(props, ownProps);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to mapDispatchToProps methods', () => {
    const tree = mapDispatchToProps(emptyProps);
    expect(tree).toMatchSnapshot();
  });
});
