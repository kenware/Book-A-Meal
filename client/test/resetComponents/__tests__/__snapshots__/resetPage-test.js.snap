// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ResetLink component renders correctly 1`] = `
<div>
  <Nav
    cart={Array []}
    component=""
    onModal={[Function]}
  />
  <div
    className="register-container"
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
    >
      <h2
        className="login-header"
      >
        Enter Your New Password
      </h2>
      <h3
        className="login-header danger text-center"
      />
      <h3
        className="login-header p-color text-center"
      />
      <form
        action="dashboard.html"
      >
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            New Password 
            <br />
            <span
              className="danger"
            >
               
            </span>
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="124gdt446776j7u"
            type="password"
          />
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="inputPassword3"
          >
            <Link
              replace={false}
              to="/login"
            >
              Login
            </Link>
                
            <Link
              replace={false}
              to="/register"
            >
              Register?
            </Link>
          </label>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
  <footer />
</div>
`;

exports[`Test ResetLink component should render messages from props 1`] = `
<div>
  <Nav
    cart={Array []}
    component=""
    onModal={[Function]}
  />
  <div
    className="register-container"
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
    >
      <h2
        className="login-header"
      >
        Enter Your New Password
      </h2>
      <h3
        className="login-header danger text-center"
      />
      <h3
        className="login-header p-color text-center"
      />
      <form
        action="dashboard.html"
      >
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            New Password 
            <br />
            <span
              className="danger"
            >
               
            </span>
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="124gdt446776j7u"
            type="password"
          />
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="inputPassword3"
          >
            <Link
              replace={false}
              to="/login"
            >
              Login
            </Link>
                
            <Link
              replace={false}
              to="/register"
            >
              Register?
            </Link>
          </label>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
  <footer />
</div>
`;

exports[`Test ResetLink component should respond to change event and change the state of the resetLink Component 1`] = `
<div>
  <Nav
    cart={Array []}
    component=""
    onModal={[Function]}
  />
  <div
    className="register-container"
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
    >
      <h2
        className="login-header"
      >
        Enter Your New Password
      </h2>
      <h3
        className="login-header danger text-center"
      />
      <h3
        className="login-header p-color text-center"
      />
      <form
        action="dashboard.html"
      >
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            New Password 
            <br />
            <span
              className="danger"
            >
              this field is required
               
            </span>
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="124gdt446776j7u"
            type="password"
          />
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="inputPassword3"
          >
            <Link
              replace={false}
              to="/login"
            >
              Login
            </Link>
                
            <Link
              replace={false}
              to="/register"
            >
              Register?
            </Link>
          </label>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
  <footer />
</div>
`;

exports[`Test ResetLink component should respond to componentWillRecievProps methods 1`] = `
<div>
  <Nav
    cart={Array []}
    component=""
    onModal={[Function]}
  />
  <div
    className="register-container"
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
    >
      <h2
        className="login-header"
      >
        Enter Your New Password
      </h2>
      <h3
        className="login-header danger text-center"
      />
      <h3
        className="login-header p-color text-center"
      />
      <form
        action="dashboard.html"
      >
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            New Password 
            <br />
            <span
              className="danger"
            >
               
            </span>
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="124gdt446776j7u"
            type="password"
          />
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="inputPassword3"
          >
            <Link
              replace={false}
              to="/login"
            >
              Login
            </Link>
                
            <Link
              replace={false}
              to="/register"
            >
              Register?
            </Link>
          </label>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
  <footer />
</div>
`;

exports[`Test ResetLink component should respond to mapDispatchToProps methods 1`] = `
Object {
  "actions": Object {
    "changePassword": [Function],
    "clearMessages": [Function],
    "getNotifications": [Function],
    "getUser": [Function],
    "loadErrorMessage": [Function],
    "loadNotifications": [Function],
    "loadSuccessMessage": [Function],
    "loadUser": [Function],
    "login": [Function],
    "refreshToken": [Function],
    "register": [Function],
    "resetLink": [Function],
    "updateProfile": [Function],
    "upgrade": [Function],
  },
}
`;

exports[`Test ResetLink component should respond to mapStateToProps methods 1`] = `
Object {
  "errorMessage": Object {
    "resetError": "Error sending link",
  },
  "successMessage": Object {
    "resetSuccess": "Link sent to user email",
  },
}
`;
